nav2_motion_primitives/src/back_up.cpp

```diff
@@ -66,46 +65,34 @@ nav2_tasks::TaskStatus BackUp::onCycleUpdate(nav2_tasks::BackUpResult & result)
   return status;
 }
 
-nav2_tasks::TaskStatus BackUp::timedBackup()
+
+nav2_tasks::TaskStatus BackUp::controlledBackup()
 {
-  // Output control command
-  geometry_msgs::msg::Twist cmd_vel;
+  auto current_odom_pose = std::make_shared<nav_msgs::msg::Odometry>();
 
-  // TODO(orduno): #423 assuming robot was moving fwd when it got stuck
-  //               fixed speed
-  cmd_vel.linear.x = -0.025;
+  if (!robot_->getOdometry(current_odom_pose)) {
+    RCLCPP_ERROR(node_->get_logger(), "Current robot odom is not available.");
+    return TaskStatus::FAILED;
+  }
+
+  geometry_msgs::msg::Twist cmd_vel;
   cmd_vel.linear.y = 0.0;
   cmd_vel.angular.z = 0.0;
-  robot_->sendVelocity(cmd_vel);
 
-// TODO(orduno): #423 fixed time
-  auto current_time = std::chrono::system_clock::now();
-  if (current_time - start_time_ >= 3s) {
-    // Stop the robot
-    cmd_vel.linear.x = 0.0;
-    robot_->sendVelocity(cmd_vel);
+  double diff_x = initial_pose_->pose.pose.position.x - current_odom_pose->pose.pose.position.x;
+  double diff_y = initial_pose_->pose.pose.position.y - current_odom_pose->pose.pose.position.y;
+  double distance = sqrt(diff_x * diff_x + diff_y * diff_y);
 
+  if (distance > abs(command_x_)) {
```

Comment:
Can we change this to `>=`? That way if someone sends a command distance of 0, we'll succeed immediately.